modules/Router.js

```diff
@@ -62,6 +62,8 @@ const Router = React.createClass({
       if (error) {
         this.handleError(error)
       } else {
+        router.location = state.location
+        router.params = state.params
```

Comment:
No, the point I’m trying to make is [you don’t need a special integrating library for React Router and Redux to work together](https://github.com/reactjs/react-router-redux/pull/387/files). There are a few problems that need to be worked out for them to work fine without any libraries in the middle:

* Fix `Link` not updating due to `connect()` optimizations (#3430)
* Make `withRouter()` more useful in deep hierarchies (#3325)
* Ensure it also works with `connect()` optimizations (#3439)

Neither of these are about putting something in the store. Rather, they are about making props from React Router easy to inject into `connect()`ed components that can read them from `ownProps` and use them for state selection. I hope this helps!